### PR TITLE
Simplify parser

### PR DESCRIPTION
**Pull request by rekado**
This is the next set of changes to simplify the parser.  The tests still pass, but the changes are less conservative.

Instead of using a LALR parser we simply recurse over the expression and build up the graph.  This allows us to get rid of nyacc.

There's still more work to be done that can build on this set of changes:

- read the expression directly instead of awkwardly converting it back and forth from a string
- don't mutate module global state (`*nodes*` etc) and just accumulate state in `atomese->graph`
- return the graph data structure directly

and finally:

- get rid of the Guile JSON fork by converting the graph record to a simple S-expression before converting it to JSON.

### CODE REVIEW

**Comment by rekado** [comment]
With the last commit the fork of Guile JSON is no longer needed.  Yay!

**Comment by Habush** [comment]
@rekado Thank you! All tests pass.

File: Dockerfile [hunk 1]
```diff
@@ -2,15 +2,9 @@ FROM xabush/annotation-scheme:0.1
...
-    cd nyacc && \
-    ./configure && \
-    make && \
-    make install
-
-WORKDIR $HOME
-RUN git clone https://github.com/Habush/guile-json.git && \
+RUN git clone https://github.com/aconchillo/guile-json && \
     cd guile-json && \
+    git reset --hard 1.2.0 && \
```
Comment (leungmanhin) [inline Dockerfile]: BTW does it really need this specific version 1.2.0, I accidentally pulled the latest 3.3.0, and it seems to work too?

**Comment by rekado** [comment]
@leungmanhin Guile JSON 3.0 changed the internal representation of values.  It uses vectors instead of lists and no longer uses hash maps IIRC.

Since the test coverage is very limited I decided to keep the changes conservative.  Upgrading to Guile JSON 3.0 should come with additions to the test suite to make sure that the JSON values the code processes have the expected type.

**Comment by leungmanhin** [comment]
Good to know, thank you @rekado

**Pull request merged**